### PR TITLE
closes #33, closes #29, closes #36

### LINKED ISSUE #29: remove item from cart

### LINKED ISSUE #33: tests for story 5

### LINKED ISSUE #36: class +controller methods for story 5
accomplish delete function

### PR DESCRIPTION
**Pull request by samanthagongora**
Adds refactor to cart necessary for delete tests and per the lesson plan. 
Adds cart delete tests.

### CODE REVIEW

File: spec/features/cart/user_can_delete_item_from_cart_spec.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
+require 'rails_helper'
+
+RSpec.feature "user visits cart" do
+  before(:each) do
+    @course1 = create(:course, id: 1)
+    @course2 = create(:course, id: 2)
+    @cart = Cart.new({"1" => 2, "2" => 3})
+  end
+
+  scenario "can delete course from cart and link back to item" do
```
Comment (DesTodo) [inline spec/features/cart/user_can_delete_item_from_cart_spec.rb]: @samanthagongora 
I had to add the following code to add the cart to session: 

    visit root_path
    click_button "Add to Cart", match: :first

**Comment by DesTodo** [comment]
@samanthagongora @yoasyo25 @thedanielvogelsang 
Sam and I are now working on the same branch for story 5. I merged my local method branch into Sam's testing branch and pushed to origin on branch: SG-story-5-tests -- this now contains all the code for story 5 from myself and Sam. 

The original pull request is asking to merge to master instead of of development; let's not merge until all bugs are fixed.

**Review by DesTodo — CHANGES_REQUESTED** [review]
1. user_adds_course_spec is broken due to adding image_tag to display image in course show. 

2. I introduced and extra step/click into the user_deletes_course_spec to add card to session; may require refactor. 

3. need to decide where we want create in cart to reroute after a course has been deleted and re-added to cart; currently set up reload individual course page, may want to set back to redirect to root/catalog; extra button may be necessary in individual course show view for ease of access (back_to_cart || back_to_list)

**Comment by samanthagongora** [comment]
user_adds_course_spec is broken due to adding image_tag to display image in course show. **resolved: test now passes**

I introduced and extra step/click into the user_deletes_course_spec to add card to session; may require refactor.
**resolved: test now passes**

need to decide where we want create in cart to reroute after a course has been deleted and re-added to cart; currently set up reload individual course page, may want to set back to redirect to root/catalog; extra button may be necessary in individual course show view for ease of access (back_to_cart || back_to_list)
**resolved:  I agree with adding individual show view to waffle. Fine as is for this iteration.**

**Review by samanthagongora — COMMENTED** [review]
user_adds_course_spec is broken due to adding image_tag to display image in course show. resolved: test now passes

I introduced and extra step/click into the user_deletes_course_spec to add card to session; may require refactor.
resolved: test now passes

need to decide where we want create in cart to reroute after a course has been deleted and re-added to cart; currently set up reload individual course page, may want to set back to redirect to root/catalog; extra button may be necessary in individual course show view for ease of access (back_to_cart || back_to_list)
resolved: I agree with adding individual show view to waffle. Fine as is for this iteration.

**Pull request merged**